From: Ondrej Jirman <megous@megous.com>
Date: Mon, 18 Oct 2021 03:55:15 +0200
Subject: arm64: dts: rk3399-pinephone-pro: Fix goodix toucscreen interrupts

The interrupt type read from the screen does not work well. It generates
constant stream of interrupts.

Change to rising edge interrupt, and enforce it in the driver via
'use-dt-irq-flags'.

Signed-off-by: Ondrej Jirman <megous@megous.com>
---
 arch/arm64/boot/dts/rockchip/rk3399-pinephone-pro.dts | 1 +
 1 file changed, 1 insertion(+)

diff --git a/arch/arm64/boot/dts/rockchip/rk3399-pinephone-pro.dts b/arch/arm64/boot/dts/rockchip/rk3399-pinephone-pro.dts
index eb3cdf4..7770275 100644
--- a/arch/arm64/boot/dts/rockchip/rk3399-pinephone-pro.dts
+++ b/arch/arm64/boot/dts/rockchip/rk3399-pinephone-pro.dts
@@ -561,6 +561,7 @@ touchscreen@14 {
 		reg = <0x14>;
 		interrupt-parent = <&gpio3>;
 		interrupts = <RK_PB5 IRQ_TYPE_EDGE_RISING>;
+		use-dt-irq-flags;
 		irq-gpios = <&gpio3 RK_PB5 GPIO_ACTIVE_HIGH>;
 		reset-gpios = <&gpio3 RK_PB4 GPIO_ACTIVE_HIGH>;
 		AVDD28-supply = <&vcc3v0_touch>;
